### PR TITLE
Add gnxi parsers

### PR DESCRIPTION
**Pull request by GerriorL**
## Description

Parser for 'show gnxi state'
Parser for 'show gnxi state detail'

## Motivation and Context
This change was originally added by https://github.com/alisharif2 but the PR grew too old to safely merge
https://github.com/CiscoTestAutomation/genieparser/pull/481

## Checklist:

- [x] I have updated the changelog.
- [ ] I have updated the documentation (If applicable).
- [x] I have added tests to cover my changes (If applicable).
- [x] All new and existing tests passed.
- [x] All new code passed compilation.

### CODE REVIEW

File: src/genie/libs/parser/iosxe/show_gnxi.py [hunk 1]
```diff
@@ -0,0 +1,342 @@
...
+        if output is None:
+            out = self.device.execute(self.cli_command)
+        else:
+            out = output
+
+        # Disabled         Down
+        # Enabled          Up
+        p_full = re.compile(r"(?P<state>Disabled|Enabled)\s*(?P<status>Down|Up)")
+
+        matches = p_full.search(out)
```
Comment (domachad) [inline src/genie/libs/parser/iosxe/show_gnxi.py:39]: I would still use a `for line in output.splitlines():` for consistency's sake

File: src/genie/libs/parser/iosxe/show_gnxi.py [hunk 2]
```diff
@@ -0,0 +1,342 @@
...
+    """ Parser for:
+            show gnxi state
+    """
+    cli_command = "show gnxi state"
+
+    def cli(self, output=None):
+        if output is None:
+            out = self.device.execute(self.cli_command)
+        else:
+            out = output
```
Comment (domachad) [inline src/genie/libs/parser/iosxe/show_gnxi.py:33]:
```suggestion
        if output is None:
            output = self.device.execute(self.cli_command)
```

File: src/genie/libs/parser/iosxe/show_gnxi.py [hunk 3]
```diff
@@ -0,0 +1,342 @@
...
+        'GNOI_CERT': "cert_management_service",
+        'GNOI_OS': "os_image_service",
+        'GNOI_RESET': "factory_reset_service"
+    }
+
+    def cli(self, output=None):
+        if output is None:
+            out = self.device.execute(self.cli_command)
+        else:
+            out = output
```
Comment (domachad) [inline src/genie/libs/parser/iosxe/show_gnxi.py:124]:
```suggestion
        if output is None:
            output = self.device.execute(self.cli_command)
```

File: src/genie/libs/parser/iosxe/show_gnxi.py [hunk 4]
```diff
@@ -0,0 +1,342 @@
...
+        p_oper_status = re.compile(r"^Oper status: (?P<oper_status>Down|Up)$")
+
+        # Supported: Supported
+        # Supported: Not supported on this platform
+        p_supported = re.compile(r"^Supported: (?P<supported>Not supported on this platform|Supported)$")
+
+        ret_dict = {}
+        current_section = None
+        current_subsection = None
+        for line in out.splitlines():
```
Comment (domachad) [inline src/genie/libs/parser/iosxe/show_gnxi.py:188]:
```suggestion
        for line in output.splitlines():
```

File: src/genie/libs/parser/iosxe/show_gnxi.py [hunk 5]
```diff
@@ -0,0 +1,342 @@
...
+            m = p_settings.match(line)
+            if m:
+                current_section = self.sections['SETTINGS']
+                continue
+            m = p_gnmi.match(line)
+            if m:
+                current_section = self.sections['GNMI']
+                continue
+            m = p_gnoi.match(line)
+            if m:
```
Comment (domachad) [inline src/genie/libs/parser/iosxe/show_gnxi.py:201]: add examples of strings that would match these patterns

File: src/genie/libs/parser/iosxe/show_gnxi.py [hunk 6]
```diff
@@ -0,0 +1,342 @@
...
+                current_subsection = self.subsections['GNOI_CERT']
+                continue
+            m = p_os_section.match(line)
+            if m:
+                current_subsection = self.subsections['GNOI_OS']
+                continue
+            m = p_reset_section.match(line)
+            if m:
+                current_subsection = self.subsections['GNOI_RESET']
+                continue
```
Comment (domachad) [inline src/genie/libs/parser/iosxe/show_gnxi.py:229]: same here

**Review by domachad — APPROVED** [review]

**Review by sjpatel21 — APPROVED** [review]

**Pull request merged**